internal/token/token.go

```diff
@@ -0,0 +1,41 @@
+package token
+
+import (
+	"os"
+
+	"github.com/google/uuid"
+	"github.com/windmilleng/wmclient/pkg/dirs"
+)
+
+const tokenPath = "token"
+
+type Token string
+
+func GetOrCreateToken(dir *dirs.WindmillDir) (Token, error) {
+	token, err := getExistingToken(dir)
+	if os.IsNotExist(err) {
+		u := uuid.New()
```

Comment:
Could we just use a random string instead? Or do you know if Google's UUID package attempt to be unguessable? The [docs](https://godoc.org/github.com/google/uuid) don't make mention of guessability, but references [RFC 4122](https://tools.ietf.org/html/rfc4122), which seems pretty explicit: "Do not assume that UUIDs are hard to guess; they should not be used as security capabilities (identifiers whose mere possession grants access), for example."

Reply:
I looked in to this and these UUIDs _are_ random strings.

google/uuid uses crypto/rand under the hood which is what we would use to generate a random string. It in turn uses `getrandom(2)` on Linux and `getentropy(2)` on macOS, which as far as I can tell from the man pages are secure/unguessable.

Furthermore google/uuid [reads 122 random bits](https://github.com/google/uuid/blob/master/version4.go#L24).